content/agent_interaction.qmd

```diff
@@ -1,12 +1,202 @@
 ---
-toc-title: '![](../imgs/cover.jpg){width=240px}<br> <h3>Agent interaction</h3>'
+toc-title: '![](../imgs/llm_as_a_judge.png){width=240px}<br> <h3>Agent interaction</h3>'
 ---
 
-# Agent interaction
+# Agent interactions
+
+In this chapter, we want to introduce multi agent systems. As a starting point, we will talk about LLM as a judge. 
+
+
+## LLM as a judge
+
+Before we introduce the concept proper, let us first describe the problem it tries to solve: 
+
+1. We generate text (be it natural language or structured output) using LLMs. 
+2. The generated text is not always correct or appropriate for our use case.
+3. We need a way to evaluate the quality of the generated text. 
+4. To do this, we have to read it.
+5. We don't have time for this.
+
+The solution to this problem is, of course, to use an LLM to read and evaluate the text. This is only fair and proper, since it was an LLM that generated the text in the first place. The generated evaluation can then be used 
+
+- to decide whether to accept or reject the generated text.
+- to improve the model itself (e.g., for fine-tuning it on the generated text and its evaluation).
+- to get an LLM to improve the text based on the evaluation.
+
+This approach is called LLM as a judge. It is a system that uses several calls to one or several LLMs to solve a problem. As such, it is an example of a multi-agent system. 
```

Comment:
```suggestion
This approach is called LLM as a judge. It is a system that uses several calls to one or several LLMs to solve a problem. As such, it can be implemented as a multi-agent system. 
```